src/uproot/writing/interpret.py

```diff
@@ -0,0 +1,306 @@
+
+    asym_errors = [x_errors_low, x_errors_high, y_errors_low, y_errors_high]
+    asym_errors_bool = [err is not None for err in asym_errors]
+
+    tgraph_type = "TGraph"
+
+    # Detecting which type of TGraph to chose
+    if any(sym_errors_bool):
+        if not all(sym_errors_bool):
+            raise ValueError("Have to specify both x_errors and y_errors")
+        if any(asym_errors_bool):
+            raise ValueError(
+                "Can's specify both symetrical errors and asymetrical errors"
+            )
+        tgraph_type = "TGraphErrors"
+
+    elif any(asym_errors_bool):
+        if not all(asym_errors_bool):
+            raise ValueError(
+                "Have to specify all: x_errors_low, x_errors_high, y_errors_low, y_errors_high"
+            )
+        if any(sym_errors_bool):
+            raise ValueError(
+                "Can's specify both symetrical errors and asymetrical errors"
+            )
+        tgraph_type = "TGraphAsymmErrors"
+
+    tobject = uproot.models.TObject.Model_TObject.empty()
+
+    tnamed = uproot.models.TNamed.Model_TNamed.empty()
+    tnamed._deeply_writable = True
+    tnamed._bases.append(tobject)
+    tnamed._members["fName"] = (
+        ""  #  Temporary name, will be overwritten by the writing process because Uproot's write syntax is ``file[name] = histogram``
+    )
+    # Constraint so user won't break TGraph naming
+    if ";" in title or ";" in xAxisLabel or ";" in yAxisLabel:
+        raise ValueError("title and xAxisLabel and yAxisLabel can't contain ';'!")
+    fTitle = f"{title};{xAxisLabel};{yAxisLabel}"
+    tnamed._members["fTitle"] = fTitle
+
+    # setting line styling
+    tattline = uproot.models.TAtt.Model_TAttLine_v2.empty()
+    tattline._deeply_writable = True
+    tattline._members["fLineColor"] = lineColor
+    tattline._members["fLineStyle"] = lineStyle
+    tattline._members["fLineWidth"] = lineWidth
+
+    # setting filling styling, does not do anything to TGraph
+    tattfill = uproot.models.TAtt.Model_TAttFill_v2.empty()
+    tattfill._deeply_writable = True
+    tattfill._members["fFillColor"] = fillColor
+    tattfill._members["fFillStyle"] = fillStyle
+
+    # setting marker styling, those are points on graph
+    tattmarker = uproot.models.TAtt.Model_TAttMarker_v2.empty()
+    tattmarker._deeply_writable = True
+    tattmarker._members["fMarkerColor"] = markerColor
+    tattmarker._members["fMarkerStyle"] = markerStyle
+    tattmarker._members["fMarkerSize"] = markerSize
+
+    if len(x) != len(y):
+        raise ValueError("Arrays x and y must have the same length!")
+    if len(x) == 0:
+        raise ValueError("Arguments and values arrays can't be emty")
+    if len(x.shape) != 1:
+        raise ValueError(f"x has to be 1D, but is {len(x.shape)}D!")
+    if len(y.shape) != 1:
+        raise ValueError(f"y has to be 1D, but is {len(y.shape)}D!")
+
+    if minY is None:
+        new_minY = min(x, default=0)
+    elif not isinstance(minY, (int, float)):
```

Comment:
```suggestion
    elif not isinstance(minY, numbers.Real):
```

and `import numbers` at the top of the file. (It's a Python standard library.) If you only check for `(int, float)`, you'll miss NumPy integers:

```python
>>> isinstance(np.int64(123), (int, float))
False
```